backend/infrahub/core/merge.py

```diff
@@ -434,12 +434,10 @@ async def merge_graph(  # pylint: disable=too-many-branches,too-many-statements
 
     async def merge_repositories(self) -> None:
         # Collect all Repositories in Main because we'll need the commit in Main for each one.
-        repos_in_main_list: list[CoreGenericRepository] = await NodeManager.query(
-            schema=InfrahubKind.REPOSITORY, db=self.db
-        )
+        repos_in_main_list: list[CoreRepository] = await NodeManager.query(schema=InfrahubKind.REPOSITORY, db=self.db)
         repos_in_main = {repo.id: repo for repo in repos_in_main_list}
 
-        repos_in_branch_list: list[CoreGenericRepository] = await NodeManager.query(
+        repos_in_branch_list: list[CoreRepository] = await NodeManager.query(
```

Comment:
This is to reflect that we're actually querying for a CoreRepository and not generic ones.